src/cpu/core_dynrec/cache.h

```diff
@@ -597,15 +605,16 @@ static void cache_init(bool enable) {
 		cache_initialized = true;
 		if (cache_blocks == NULL) {
 			// allocate the cache blocks memory
-			cache_blocks=(CacheBlockDynRec*)malloc(CACHE_BLOCKS*sizeof(CacheBlockDynRec));
+			cache_blocks = (CacheBlock *)malloc(CACHE_BLOCKS *
+			                                    sizeof(CacheBlock));
 			if(!cache_blocks) E_Exit("Allocating cache_blocks has failed");
-			memset(cache_blocks,0,sizeof(CacheBlockDynRec)*CACHE_BLOCKS);
+			memset(cache_blocks, 0, sizeof(CacheBlock) * CACHE_BLOCKS);
 			cache.block.free=&cache_blocks[0];
 			// initialize the cache blocks
 			for (i=0;i<CACHE_BLOCKS-1;i++) {
-				cache_blocks[i].link[0].to=(CacheBlockDynRec *)1;
-				cache_blocks[i].link[1].to=(CacheBlockDynRec *)1;
-				cache_blocks[i].cache.next=&cache_blocks[i+1];
+				cache_blocks[i].link[0].to = (CacheBlock *)1;
+				cache_blocks[i].link[1].to = (CacheBlock *)1;
```

Comment:
Very interesting / strange :-)  The number 1 is being cast to a CacheBlock pointer-type?

Reply:
Yeah, this is a bit weird… It seems like this value is used only for emulating conditional jumps. ATM I don't understand why pointer with value 1 is used (and digging in history does not help - this was introduced with initial commit creating dynrec).